lib/Sema/TypeCheckAvailability.cpp

```diff
@@ -1186,7 +1186,7 @@ static void fixAvailabilityForDecl(SourceRange ReferenceRange, const Decl *D,
         << OriginalIndent;
   }
 
-  TC.diagnose(ReferenceRange.Start, diag::availability_add_attribute,
+  TC.diagnose(InsertLoc, diag::availability_add_attribute,
```

Comment:
I think it would be nicer to put this at the location of `D` instead. What do you think? Note that you can just pass a Decl here and the diagnostic engine will automatically get its location; no need to call `getLocation` or anything.

(This is the location of the caret in the diagnostic.)